### PR TITLE
Add gitleaks section to contrib doc

### PR DESCRIPTION
**Pull request by eranha**
Pushing to github is a form of publication, especially when using a public repo. It is a good idea to use a hook to check for secrets before pushing code.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,6 +42,11 @@ Before getting started, you should install some developer tools. These are not
 required to deploy Conjur but they will let you develop using a standardized,
 expertly configured environment.
 
+
+##### Preventing Secret Leaks
```
Comment (orenbm) [inline CONTRIBUTING.md:46]: looks out of place to me. how can it be before you install git?

also - why are there 4 `#`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -42,6 +42,11 @@ Before getting started, you should install some developer tools. These are not
...
+Pushing to github is a form of publication, especially when using a public repo. It is a good idea to use a hook to check for secrets before pushing code.
+Follow this [link](https://github.com/cyberark/community/blob/master/Conjur/conventions/git-tips-and-tricks.md#preventing-leaks) on how to go about configuring git check for secrets before every push.
```
Comment (izgeri) [inline CONTRIBUTING.md:48]: this is in a weird spot, it's inserted between the prerequisite intro sentence and the list of prerequisites, and it has 5 hashtags when the parent section has 2 (so this subsection should just have 3)

can you please:
- move it down below the prerequisites
- change the number of `#` in the title
- add it to the TOC

**Review by izgeri — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -50,6 +50,11 @@ expertly configured environment.
 [get-git]: https://git-scm.com/downloads
 [get-docker-compose]: https://docs.docker.com/compose/install
 
+### Preventing Secret Leaks
```
Comment (orenbm) [inline CONTRIBUTING.md:53]: i would change to `Prevent secret leaks` to maintain the language of this doc

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -50,6 +50,11 @@ expertly configured environment.
...
+Pushing to github is a form of publication, especially when using a public repo. It is a good idea to use a hook to check for secrets before pushing code.
+Follow this [link](https://github.com/cyberark/community/blob/master/Conjur/conventions/git-tips-and-tricks.md#preventing-leaks) to learn how to configure git checks for secrets before every push.
+
+
```
Comment (orenbm) [inline CONTRIBUTING.md:57]: remove empty line

**Review by orenbm — APPROVED** [review]

**Review by izgeri — DISMISSED** [review]

**Review by nessiLahav — APPROVED** [review]
LGTM

**Review by nessiLahav — APPROVED** [review]
LGTM

**Review by nessiLahav — APPROVED** [review]

**Pull request merged**